### ISSUE TITLE
[Feature Request]: Checks

### ISSUE THREAD

**Issue by BlueTree242** [A248/LibertyBans#88]
### I have confirmed that ...

- [X] There is no duplicate feature request.

### Description

so, on a smp server, a mod can be annoying and ban innocent players, how could we prevent that??

Here is how, the plugin does checks before punishing a player, still not clear, ill give more details.

This is an example list of checks
- Anticheat detected before
- Reported by a player before
- Player has sent too many messages racently
- Player swore alot
- Player placed alot of redstone blocks (for smps)
- Player tried to ping everyone on discordsrv
- Player is not brand new, but this doesn't seem so good

I am not requesting the stuff above, the above should be added by other plugins via the api

For example

```java
registerCheck("cheated-before", player -> {
 If (anticheat.playerDidCheat(player)) return true;
else return false
});```

The first argument is the thing used for bypass perms, like `libertybans.bypasscheck.cheated-before`

This would make this plugin more powerful than any other plugin ever.


Result for owners:

Less chance of false punishments

**Comment by KoxSosen** [comment]
I think your feature request is interesting, but sounds like something that should be done by another plugin. Adding many types of checks can be boilerplate for those who doesn't need it, and can hurt preformance, not to speak about the maintance costs.

LibertyBans is kinda like a tool, a utility other plugins should hook into. I'd love to see anticheats making addons or implementing LibertyBans support, I think if you are looking for something like that for as an example..You can code it! ^_^

**Comment by KoxSosen** [comment]
> so, on a smp server (and other servers too), a mod can be annoying and ban innocent players, how could we prevent that??
Remove the mod :p

**Comment by BlueTree242** [comment]
> I think your feature request is interesting, but sounds like something that should be done by another plugin. Adding many types of checks can be boilerplate for those who doesn't need it, and can hurt preformance, not to speak about the maintance costs.
> 
> LibertyBans is kinda like a tool, a utility other plugins should hook into. I'd love to see anticheats making addons or implementing LibertyBans support, I think if you are looking for something like that for as an example..You can code it! ^_^

I didn't mention it should be off by default sorry. This beeing built in the plugin can get the plugin instantly on top

**Comment by BlueTree242** [comment]
> > so, on a smp server (and other servers too), a mod can be annoying and ban innocent players, how could we prevent that??
> 
> Remove the mod :p

Well yeah doesnt work on big servers with those busy owners who have no time to see if players are innocent or can't proof they are

**Comment by BlueTree242** [comment]
No one would hook into an addon so it cant be like that

**Comment by BlueTree242** [comment]
About performance, some plugins checks can be slow, just adding a list of disabled checks would help

**Comment by KoxSosen** [comment]
> > > so, on a smp server (and other servers too), a mod can be annoying and ban innocent players, how could we prevent that??
> > 
> > 
> > Remove the mod :p
> 
> Well yeah doesnt work on big servers with those busy owners who have no time to see if players are innocent or can't proof they are

Good point, but from my experience large serburs usually have a staff team, and instead of relying on anticheat bans, they usually use the anticheat alerts as a way to inform staff members about potencial cheaters.

**Comment by BlueTree242** [comment]
> > > > so, on a smp server (and other servers too), a mod can be annoying and ban innocent players, how could we prevent that??
> > > 
> > > 
> > > Remove the mod :p
> > 
> > 
> > Well yeah doesnt work on big servers with those busy owners who have no time to see if players are innocent or can't proof they are
> 
> Good point, but from my experience large serburs usually have a staff team, and instead of relying on anticheat bans, they usually use the anticheat alerts as a way to inform staff members about potencial cheaters.

Yes for anticheat, it would check if they cheated before as it would be a proof this is a legit ban, note these checks only happen on small staff like trial or mod

**Comment by BlueTree242** [comment]
The feature may be useless for some people, but this can make the plugin on top even if low server use it

**Comment by A248** [comment]
I definitely appreciate the proposal for such an expansive feature – there has been no other feature request to-date which is as ambitious as this one (I consider that a good thing).

However, this does seem like a feature which would belong in another plugin. Nevertheless, LibertyBans should make this feature possible using the API – that is, LibertyBans should provide enough API features to enable another plugin to build this kind of system on top of it.

If there was a plugin, either existing or in-progress, that provided such a system of checks, then we could see whether the LibertyBans API is adequate to support the checks system. If the API is inadequate, that would be good reason to add API features.

Regarding the popularity of LibertyBans, I think that if another plugin provided this system of checks, and was built on top of LibertyBans, that would similarly boost LibertyBans' popularity, and in a manner compatible with the design and scope of LibertyBans. That design and scope being to avoid features, where possible, which are tangential or additional to a basic system of punishment administration and enforcement. If a feature can be adequately and efficiently implemented in another plugin, then it should probably not be included in LibertyBans.

**Comment by lokka30** [comment]
I agree with A2 here - this seems best fit for an 'addon'-type plugin.
